agent/engine/task_manager.go

```diff
@@ -420,15 +438,19 @@ func (mtask *managedTask) progressContainers() {
 	// complete, but keep reading events as we do.. in fact, we have to for
 	// transitions to complete
 	mtask.waitForContainerTransitions(transitions, transitionChange, transitionChangeContainer)
-	log.Debug("Done transitioning all containers for task", "task", mtask.Task)
+	seelog.Debug("Done transitioning all containers for task %v", mtask.Task)
 
-	if mtask.UpdateStatus() {
-		log.Debug("Container change also resulted in task change")
+	// update the task status
+	changed := mtask.UpdateStatus()
```

Comment:
Why split this into two lines? `UpdateStatus` is just a boolean without side effects right?